src/main/scala/com/databricks/spark/xml/parsers/StaxXmlParser.scala

```diff
@@ -234,16 +234,15 @@ private[xml] object StaxXmlParser {
       options: XmlOptions,
       rootAttributes: Array[Attribute] = Array.empty): Row = {
     val row = new Array[Any](schema.length)
+    val nameToIndex = schema.map(_.name).zipWithIndex.toMap
```

Comment:
Would you mind if I ask to elaborate this change?

Reply:
@HyukjinKwon if we hava attributes, we need them firstly before reading values.
before we have  if we hava attributes, and don't have values -> don't read them. I fix it